### PR TITLE
fix(typeahead): run validators manually on select

### PR DESCRIPTION
**Pull request by sagens42**
fix for #3311

### CODE REVIEW

File: src/typeahead/typeahead.js [hunk 1]
```diff
@@ -251,6 +251,11 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
         $setModelValue(originalScope, model);
         modelCtrl.$setValidity('editable', true);
 
+        if (modelCtrl.$$lastCommittedViewValue === modelCtrl.$viewValue) {
+          // angular will check if model value changed to run validators, so we need to run it manually
+          modelCtrl.$$runValidators(undefined, modelCtrl.$modelValue, modelCtrl.$viewValue, angular.noop);
```
Comment (chrisirhc) [inline src/typeahead/typeahead.js]: These are undocumented API. Is there any way to this with the documented API (prefixed with a single `$`)?

Comment (sagens42) [inline src/typeahead/typeahead.js]: Actually I think it's enough to run `$setValidity` for `parse` key

File: src/typeahead/test/typeahead.spec.js [hunk 2]
```diff
@@ -625,6 +625,18 @@ describe('typeahead tests', function () {
...
+    it('issue #3311 - should trigger validation if filter value equal to list value', function() {
+
+      var element = prepareInputEl('<form name="form"><input name="input" ng-model="result" typeahead="item for item in source | filter:$viewValue"></form>');
+
+      changeInputValueTo(element, 'foo');
+
+      var match = $(findMatches(element)[0]).find('a')[0];
+      $(match).click();
+
+      expect($scope.form.input.$error.parse).toBeUndefined();
```
Comment (wesleycho) [inline src/typeahead/test/typeahead.spec.js]: You should add a check for the $error before the click as well.

Comment (sagens42) [inline src/typeahead/test/typeahead.spec.js]: `parsed` error was introduced in 1.3, so it's impossible to test it in current environment ( angular 1.2.16 )

Comment (rvanbaalen) [inline src/typeahead/test/typeahead.spec.js]: @sagens42 we've updated the project to 1.3 recently

**Comment by sagens42** [comment]
Fixed in #3166 Closing

**Comment by katerberg** [comment]
It looks like this is still an issue with the new `$validators`. If I have a custom validator added and I type the exact item in a list, it does not fire that validator again after selecting an item. Should I create a new issue for this?

**Comment by icfantv** [comment]
@katerberg - yes, please.  [Here are the instructions](https://github.com/angular-ui/bootstrap/blob/master/CONTRIBUTING.md#you-think-youve-found-a-bug) for doing so, thanks.

**Comment by katerberg** [comment]
@icfantv Created #5136. Let me know if I am missing something since this is my first time contributing a bug to this. Thanks for pointing me in the right direction!

**Pull request closed without merging**